### PR TITLE
Display rlimits type and signal number

### LINKED ISSUE #774: Missing rlimits and signals in crit decode output
While trying to explore `core-<pid>.img` using `crit decode` command, I realized rlimits output does not show the type of resource and signal information does not include the signal number.
I know the rlimit type and signal number can be retrieved by checking the index of the entry in the list, but still it is cumbersome.
Current output for rlimits:
```
                "rlimits": {
                    "rlimits": [
                        {   
                            "cur": 18446744073709551615,
                            "max": 18446744073709551615
                        },
                        {   
                            "cur": 18446744073709551615,
                            "max": 18446744073709551615
                        },
                        {   
                            "cur": 18446744073709551615,
                            "max": 18446744073709551615
                        },
```
For signals:
```
                "sigactions": [
                    {   
                        "sigaction": "0x0",
                        "flags": "0x0",
                        "restorer": "0x0",
                        "mask": "0x0",
                        "compat_sigaction": false
                    },
                    {   
                        "sigaction": "0x559a4cebd200",
                        "flags": "0x4000000",
                        "restorer": "0x7f38fe88ef20",
                        "mask": "0xfffffffe7ffbfeff",
                        "compat_sigaction": false
                    },
                    {   
                        "sigaction": "0x0",
                        "flags": "0x4000000",
                        "restorer": "0x7f38fe88ef20",
                        "mask": "0xfffffffe7ffbfeff",
                        "compat_sigaction": false
                    },
                    {
                        "sigaction": "0x0",
                        "flags": "0x0",
                        "restorer": "0x0",
                        "mask": "0x0",
                        "compat_sigaction": false
                    },
```
It would be useful to interpret the output if we include the name of resource (eg RLIMIT_AS, RLIMIT_CPU, RLIMIT_CORE) and signal number in the output.

### PR DESCRIPTION
**Pull request by ashu-mehra**
Display rlimits type and signal number when decoding
core-<pid>.img file using crit

Fixes #774 
Signed-off-by: Ashutosh Mehra <asmehra1@in.ibm.com>

### CODE REVIEW

**Comment by ashu-mehra** [comment]
Sample output after this PR:

rlimits info:
```
                    "rlimits": [
                        {
                            "cur": 18446744073709551615,
                            "max": 18446744073709551615,
                            "rlimit_type": "RLIMIT_CPU"
                        },
                        {
                            "cur": 18446744073709551615,
                            "max": 18446744073709551615,
                            "rlimit_type": "RLIMIT_FSIZE"
                        },
```

Signals info:
```
                "sigactions": [
                    {
                        "sigaction": "0x0",
                        "flags": "0x0",
                        "restorer": "0x0",
                        "mask": "0x0",
                        "compat_sigaction": false,
                        "signal": 1
                    },
                    {
                        "sigaction": "0x5573f4d48200",
                        "flags": "0x4000000",
                        "restorer": "0x7fae27d6df20",
                        "mask": "0xfffffffe7ffbfeff",
                        "compat_sigaction": false,
                        "signal": 2
                    },
```

For signals the signal number does not strictly correspond to its entry index in in the list, as couple of signals - SIGSTOP and SIGKLL - are not dumped. Therefore we need to include the signal number in the output.
Moreover, the mapping from signal number to signal name varies across architectures, and python does not have standard API to access the signal name for a given signal number, so I didn't bother much about displaying the signal name. I guess signal number itself is sufficient enough.

File: images/rlimit.proto [hunk 1]
```diff
@@ -1,6 +1,9 @@
 syntax = "proto2";
 
+import "opts.proto";
+
 message rlimit_entry {
 	required uint64		cur = 1;
 	required uint64		max = 2;
+	optional uint32		rlimit_type = 3 [(criu).dict = "rlimit"];
```
Comment (avagin) [inline images/rlimit.proto]: This field isn't required to restore a process, so it should not be in the image file. You can use an array index to determine a type.

**Comment by avagin** [comment]
I merged this pr by mistake. It has been reverted.

**Comment by ashu-mehra** [comment]
@avagin do you want me to create a new PR for this?

**Comment by avagin** [comment]
@ashu-mehra yes. I'm sorry for breaking this one.

**Pull request merged**